src/main/java/com/wseemann/ecp/core/ECPRequest.kt

```diff
@@ -33,31 +38,16 @@ internal abstract class ECPRequest<T>(private val url: String) {
                 val request = DiscoveryRequest(url)
                 return ECPResponse(generateResponseData(request.send().data, getParser()))
             } else {
-                val okHttpClient = OkHttpClient.Builder()
-                        .connectTimeout(6000, TimeUnit.MILLISECONDS)
-                        .readTimeout(6000, TimeUnit.MILLISECONDS)
-                        .build()
-
-                val body = if (getMethod() == "POST") "".toRequestBody() else null
-
-                val request = Request.Builder()
-                        .addHeader("User-Agent", "Roku-ECP-Wrapper-Kotlin")
-                        .url(url.toHttpUrl())
-                        .method(getMethod(), body)
-                        .build()
-
-                val call = okHttpClient.newCall(request)
-                val response = call.execute()
-
-                val responseBody = response.body
-
-                if (responseBody != null) {
-                    val body = responseBody.bytes()
-                    debug("ECP request response: " + String(body))
-                    return ECPResponse(generateResponseData(body, getParser()))
-                } else {
-                    return null
+                val httpManager = HttpManager.getInstance()
+
+                if (http_request == null) {
```

Comment:
Can you explain the purpose of `http_request` here? Wouldn't this cause caching of every request regardless of the `useRequestCache` flag because once the value is set to `true` then:

 http_request = httpManager.buildRequest(url, getMethod(), useRequestCache)

is never executed?